src/procstat.c

```diff
@@ -1212,19 +1212,39 @@ static ssize_t procstat_fmt_u32_percentile(void *object, uint64_t arg, char *buf
 	return procstat_format_u32_decimal(data_ptr, 0, buffer, length);
 }
 
-
-void procstat_histogram_u32_add_point(struct procstat_histogram_u32 *series, uint32_t value)
+bool is_reset_histogram(struct procstat_histogram_u32 *series)
 {
-	unsigned reset;
+	unsigned reset = 0;
+	uint64_t reset_interval, time_since_last_reset;
+	bool reset_interval_expired = false;
 
-	reset = __atomic_load_n(&series->reset, __ATOMIC_RELAXED);
-	if (reset) {
+	struct timespec cur_time;
```

Comment:
@alexs-lb this is actually code duplication except for the type of series all is the same.
I suggest you create a a new struct called interval_info with the following fields
uint64_t                                reset_interval;
uint64_t 				last_reset_time;

Then you embed this struct in the series and histogram structures and define one  is_reset function that will get a ptr to this struct.  so you actually call it like that:
is_reset(&series->interval_info)

Reply:
Agree, this is a good suggestion.